sandbox/host/src/sandbox/wasmer_backend/cache.rs

```diff
@@ -0,0 +1,211 @@
+}
+
+// The cached_modules function provides thread-safe access to the CACHED_MODULES static.
+fn lru_cache() -> &'static CachedModules {
+    static CACHED_MODULES: OnceLock<CachedModules> = OnceLock::new();
+    CACHED_MODULES.get_or_init(|| Mutex::new(LRUCache::default()))
+}
+
+fn fs_cache() -> Result<FileSystemCache, CachedModuleErr> {
+    static CACHE_DIR: OnceLock<TempDir> = OnceLock::new();
+
+    // Try to load from tempfile cache
+    let cache_path = CACHE_DIR
+        .get_or_init(|| {
+            tempfile::tempdir().expect("Cannot create temporary directory for wasmer caches")
+        })
+        .path();
+    log::trace!("Wasmer sandbox cache dir is: {cache_path:?}");
+
+    FileSystemCache::new(cache_path).map_err(|_| FileSystemErr)
+}
+
+pub fn get_cached_module(
+    wasm: &[u8],
+    store: &sandbox_wasmer::Store,
+) -> core::result::Result<Module, CachedModuleErr> {
+    let mut lru_lock = lru_cache().lock().expect("CACHED_MODULES lock fail");
+
+    let maybe_module = lru_lock.find(|x| x.wasm == wasm);
+
+    // Try to load from LRU cache first
+    if let Some(CachedModule {
+        serialized_module, ..
+    }) = maybe_module
+    {
+        // SAFETY: Module inside LRU cache cannot be corrupted.
+        let module = unsafe {
+            Module::deserialize_unchecked(store, serialized_module.as_slice())
+                .expect("module in LRU cache is valid")
+        };
+        Ok(module)
+    } else {
+        let code_hash = Hash::generate(wasm);
+
+        let fs_cache = fs_cache()?;
+        let serialized_module = fs_cache
+            .load(code_hash)
+            .map_err(|_| CacheMissErr(fs_cache.clone(), code_hash))?;
+
+        lru_cache()
+            .lock()
+            .expect("CACHED_MODULES lock fail")
+            .insert(CachedModule {
+                wasm: wasm.to_vec(),
+                serialized_module: serialized_module.clone(),
+            });
+
+        // SAFETY: We trust the module in FS cache.
+        let module = unsafe {
+            Module::deserialize(store, serialized_module).map_err(|_| {
+                // Module in FS cache is corrupted, remove it.
+                fs_cache.remove_key(code_hash);
+                CacheMissErr(fs_cache, code_hash)
+            })?
+        };
+
+        Ok(module)
+    }
+}
+
+pub fn try_to_store_module_in_cache(
+    mut fs_cache: FileSystemCache,
+    code_hash: Hash,
+    wasm: &[u8],
+    module: &Module,
+) {
+    // NOTE: `From<Bytes> to Vec<u8>` is zero cost.
+    let serialized_module: Vec<_> = module
+        .serialize()
+        .expect("module should be serializable")
+        .into();
+
+    // Store module in LRU cache
+    let res = lru_cache()
+        .lock()
+        .expect("CACHED_MODULES lock fail")
+        .insert(CachedModule {
+            wasm: wasm.to_vec(),
+            serialized_module: serialized_module.clone(),
+        });
+    log::trace!("Store module in LRU cache with result: {:?}", res.is_some());
```

Comment:
maybe just like that?
```suggestion
    log::trace!("Store module in LRU cache with result: {res:?}");
```

Reply:
It will has additional useless info then?

Reply:
Removed misleading result in the log message